### PR TITLE
feat: a yes or no question filter

### PR DESCRIPTION
**Pull request by wwydmanski**

### CODE REVIEW

**Comment by wwydmanski** [comment]
The build error is caused by a pip problem:
`ERROR: Could not install packages due to an OSError: [Errno 2] No such file or directory: '/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/importlib_metadata-4.6.0.dist-info/METADATA' `

Is there something I should do to circumvent that?

**Comment by aadesh11** [comment]
> The build error is caused by a pip problem:
> `ERROR: Could not install packages due to an OSError: [Errno 2] No such file or directory: '/opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/importlib_metadata-4.6.0.dist-info/METADATA' `
> 
> Is there something I should do to circumvent that?


Thanks for reporting it. Could you please take the master pull and try again.

**Comment by kaustubhdhole** [comment]
@wwydmanski thanks for your changes! Could you please resolve conflicts - you won't need to make any changes in the action files now from your side.

File: filters/yesno_question/filter.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        TaskType.TEXT_CLASSIFICATION,
+        TaskType.TEXT_TO_TEXT_GENERATION,
+        TaskType.QUESTION_ANSWERING,
+        TaskType.QUESTION_GENERATION,
+    ]
+    languages = ["en"]
+
+    def __init__(self):
+        super().__init__()
+        self.nlp = spacy.load("en_core_web_sm")
```
Comment (kaustubhdhole) [inline filters/yesno_question/filter.py:24]: You can use the common spacy.

File: filters/yesno_question/filter.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+        doc = self.nlp(question)
+        token = doc[0]
+        if token.pos_ != "AUX":
+            return False
+
+        return True
+
+
+if __name__ == "__main__":
```
Comment (kaustubhdhole) [inline filters/yesno_question/filter.py:44]: This can be commented.

**Comment by wwydmanski** [comment]
I'm not sure what is the reason for the test failure. Everything is working as it should when I do a clean download of the PR. 
Do you have an idea what might be causing it?

**Comment by kaustubhdhole** [comment]
Seems the following test case is failing: 

>           assert (
                    output == test["outputs"]
            ), f"Executing {test['class']} The filter should return {test['outputs']} for the inputs {test['inputs']}"
E           AssertionError: Executing YesNoQuestionFilter The filter should return True for the inputs {'context': '', 'question': "Wasn't she angry when you told her about the accident?", 'answers': []}
E           assert False == True

File: filters/yesno_question/filter.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
+from typing import List, Tuple
+
+import spacy
+
+from initialize import spacy_nlp
+from interfaces.QuestionAnswerOperation import QuestionAnswerOperation
+from tasks.TaskTypes import TaskType
+
+"""
+A filter on text length (number of tokens).
```
Comment (sebastianGehrmann) [inline filters/yesno_question/filter.py:10]: This doesn't seem correct.

File: filters/yesno_question/filter.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+        rest_of_sentence = " ".join(map(str, doc[verb_position:]))
+        if "or" in rest_of_sentence:
+            # This is a question about an alternative
+            return False
+
+        return True
+
+
+# if __name__ == "__main__":
```
Comment (sebastianGehrmann) [inline filters/yesno_question/filter.py:59]: Get rid of the commented-out part?

**Pull request merged**